geps/gep-1867/index.md

```diff
@@ -75,43 +75,74 @@ For example, if I wanted to represent a `version` field and change that to trigg
-  // A Gateway may provide configuration for the same values; as all fields in GatewayInfrastructure are implementation specific,
-  // the merging logic between these is as well. However, the GatewayClass is generally expected to be providing defaults
-  // rather than overrides.
-  Infrastructure GatewayClassInfrastructure `json:"infrastructure"`
-  // ...
-}
-
 type GatewayInfrastructure struct {
-  // ParametersRef provides a arbitrary implementation-specific configuration for
-  // fields not expressed directly in this struct.
-  // This follows the same semantics as GatewayClass's ParametersRef, but lives on the Gateway.
-  ParametersRef ParametersReference
+  // Labels that SHOULD be applied to any resources created in response to this Gateway.
+	//
+	// For implementations creating other Kubernetes objects, this should be the `metadata.labels` field on resources.
+	// For other implementations, this refers to any relevant (implementation specific) "labels" concepts.
+	//
+	// An implementation may chose to add additional implementation-specific labels as they see fit.
+	//
+	// Support: Extended
+	//
+	// +optional
+	// +kubebuilder:validation:MaxProperties=8
+	Labels map[AnnotationKey]AnnotationValue `json:"labels,omitempty"`
+
+	// Annotations that SHOULD be applied to any resources created in response to this Gateway.
+	//
+	// For implementations creating other Kubernetes objects, this should be the `metadata.annotations` field on resources.
+	// For other implementations, this refers to any relevant (implementation specific) "annotations" concepts.
+	//
+	// An implementation may chose to add additional implementation-specific annotations as they see fit.
+	//
+	// Support: Extended
+	//
+	// +optional
+	// +kubebuilder:validation:MaxProperties=8
+	Annotations map[AnnotationKey]AnnotationValue `json:"annotations,omitempty"`
+
+	// ParametersRef is a reference to a resource that contains the configuration
+	// parameters corresponding to the Gateway. This is optional if the
+	// controller does not require any additional configuration.
+	//
+	// This follows the same semantics as GatewayClass's `parametersRef`, but on a per-Gateway basis
+	//
+	// The Gateway's GatewayClass may provide its own `parametersRef`. When both are specified,
+	// the merging behavior is implementation specific.
+	// It is generally recommended that GatewayClass provides defaults that can be overridden by a Gateway.
+	//
+	// Support: Implementation-specific
+	//
+	// +optional
+	ParametersRef *LocalParametersReference `json:"parametersRef,omitempty"`
 }
 
 type GatewayClassInfrastructure struct {
 }
 ```
 
+!!! warning
+    Modifying pod labels via `infrastructure.labels` could result in data plane pods being replaced (depending on the implementation). The way in which the pods are replaced is implementation specific and may result in downtime. Implementations should document how they handle changes to `infrastructure.labels`.
```

Comment:
I think something like this could likely belong in the spec for infrastructure.labels as well. Here's a rough starting point, feel free to change:

"If an implementation maps these labels to Pods, or any other resource that would need to be recreated when labels change, it SHOULD clearly warn about this behavior in documentation."

Reply:
@keithmattix not sure if you intended to add this to spec, but I actually meant that it could be useful to add a comment like that to the spec somewhere around here: https://github.com/kubernetes-sigs/gateway-api/blob/24f73922acf501b547e0f183c8f8066eb6fbb045/apis/v1/gateway_types.go#L707-L720

Reply:
Ah good catch; I had the markdown open on the API section, and my eyes told my brain I was editing a Go file 🤦🏾 Should be fixed no!